### PR TITLE
fastfetch: 1.12.2 -> 2.0.0, yyjson: init at 0.7.0

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes

Update `fastfetch` to 2.0.0 and package its new `yyjson` dependency.

https://github.com/fastfetch-cli/fastfetch/blob/2.0.0/CHANGELOG.md
https://github.com/ibireme/yyjson

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Henry-Hiles — APPROVED** [review]
LGTM

File: pkgs/development/libraries/yyjson/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+}:
+
+stdenv.mkDerivation rec {
```
Comment (Gerg-L) [inline pkgs/development/libraries/yyjson/default.nix:7]: Maybe use `finalAttrs` to be more verbose

File: pkgs/development/libraries/yyjson/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    hash = "sha256-Cz8K+cWuDpoMY6d+ecuOvXIMc4wtx15LLvxpFibkNyw=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  meta = with lib; {
+    description = "The fastest JSON library in C";
+    homepage = "https://github.com/ibireme/yyjson";
```
Comment (Gerg-L) [inline pkgs/development/libraries/yyjson/default.nix:24]: This can be inherited from `src`

**Review by Gerg-L — COMMENTED** [review]
Two nitpicks

**Comment by Gerg-L** [comment]
@khaneliman can you test on Darwin when you got a chance

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 249428` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fastfetch</li>
  </ul>
</details>

**Review by khaneliman — APPROVED** [review]
LGTM works on aarch64-Darwin.

**Review by Gerg-L — APPROVED** [review]
LGTM

**Review by khaneliman — APPROVED** [review]

File: pkgs/development/libraries/yyjson/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    hash = "sha256-Cz8K+cWuDpoMY6d+ecuOvXIMc4wtx15LLvxpFibkNyw=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  meta = with lib; {
+    description = "The fastest JSON library in C";
+    homepage = finalAttrs.src.meta.homepage;
```
Comment (AndersonTorres) [inline pkgs/development/libraries/yyjson/default.nix:24]: It is preferrable to "expand" this instead.
But I will not pursue this today.

**Review by AndersonTorres — APPROVED** [review]
LGeTM

**Review by Gerg-L — APPROVED** [review]

**Review by khaneliman — APPROVED** [review]

File: pkgs/tools/misc/fastfetch/default.nix [hunk 4]
```diff
@@ -40,15 +42,23 @@
...
   src = fetchFromGitHub {
     owner = "fastfetch-cli";
     repo = "fastfetch";
     rev = finalAttrs.version;
-    hash = "sha256-l9fIm7+dBsOqGoFUYtpYESAjDy3496rDTUDQjbNU4U0=";
+    hash = "sha256-mXbkzPlX1OsK+ahUSJWktV5D7Mo2zkhXgXP54QjbIR4=";
   };
 
+  patches = [
+    # Don't fetch yyjson.
```
[the remark this answers is not in the document]
Comment (FedericoSchonborn) [reply pkgs/tools/misc/fastfetch/default.nix:55]: Looks like the next version will be [vendoring `yyjson` instead](https://github.com/fastfetch-cli/fastfetch/tree/dev/src/3rdparty/yyjson), I opened https://github.com/fastfetch-cli/fastfetch/issues/525.

**Comment by CarterLi** [comment]
* Why Windows logo was displayed? Did you set the logo explicitly?
* Are all terminals in nixos are wrapped as .*-wrapped? Why?

**Comment by FedericoSchonborn** [comment]
> * Why Windows logo was displayed? Did you set the logo explicitly?

Yes.

File: pkgs/development/libraries/yyjson/default.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/development/libraries/yyjson/default.nix:24]: this is a bad practice indeed. Nixpkgs should not rely on "macro expansion".

**Comment by Gerg-L** [comment]
> * Are all terminals in nixos are wrapped as .*-wrapped? Why?
`wrapProgram` which moves the binary for the terminal to `.*-wrapped` and replaces it with a shell script which exports env vars and launches the program with extra flags

**Comment by AndersonTorres** [comment]
>     * Are all terminals in nixos are wrapped as .*-wrapped? Why?

Sometimes such terminal-emulators call libraries, especially GTK and Qt, that rely on a bunch of environment variables. These wrappers set that env vars.

**Comment by CarterLi** [comment]
How can I find the original binaries?

**Comment by Gerg-L** [comment]
> How can I find the original binaries?

$TERM should work in most cases

File: pkgs/tools/misc/fastfetch/default.nix (hunk 4 above)
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/tools/misc/fastfetch/default.nix:55]: It was upstreamed! Please update the code - now using https://github.com/fastfetch-cli/fastfetch/commit/c7f0172b256ea3f29e6e2ef25cd43fcc2c12614e as patch source

**Pull request merged**